atlas_anndata/funcs.py

```diff
@@ -1461,13 +1477,41 @@ def calculate_summary_stats(adata, obs, matrix="normalised"):
     ...    [ marker_grouping ],
     ...    matrix='normalised')
     Calculating summary stats for normalised matrix, cell groups defined by ['louvain_resolution_0.7']
+    ..louvain_resolution_0.7
+    Completed summary stats calculation
+    >>> adata.varm['mean_normalised_louvain_resolution_0.7'] # doctest:+ELLIPSIS +NORMALIZE_WHITESPACE
+                                 0           1
+    ENSDARG00000000001    5.210635    4.868471
+    ENSDARG00000000002    5.107658    0.254911
+    ENSDARG00000000018  240.056006  319.279625
+    ENSDARG00000000019   34.447536    9.438827
+    ENSDARG00000000068   33.625522    1.097745
+    ...
+    >>> adata.varm['median_normalised_louvain_resolution_0.7'] # doctest:+ELLIPSIS +NORMALIZE_WHITESPACE
+                             0           1
+    ENSDARG00000000001    0.000000    0.000000
+    ENSDARG00000000002    0.000000    0.000000
+    ENSDARG00000000018  141.642776  159.815002
+    ENSDARG00000000019    0.000000    0.000000
+    ENSDARG00000000068    0.000000    0.000000
+    ...
     """
 
     print(
         f"Calculating summary stats for {matrix} matrix, cell groups defined"
         f" by {obs}"
     )
+
+    def matrix_summary(matrix, summary_type):
+        if summary_type == "mean":
+            return matrix.mean(axis=0, dtype=np.float64)
+        else:
+            return skl.utils.sparsefuncs.csc_median_axis_0(
+                sp.sparse.csc_matrix(matrix)
```

Comment:
I would suggest to add this if you keep the `dtype` for the mean:
```suggestion
                sp.sparse.csc_matrix(matrix, dtype=np.float64)
```